core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/datarepo/HttpDataRepoDAO.scala

```diff
@@ -25,6 +28,16 @@ class HttpDataRepoDAO(dataRepoInstanceName: String, dataRepoInstanceBasePath: St
 
   override def getSnapshot(snapshotId: UUID, accessToken: OAuth2BearerToken): SnapshotModel = {
     // future enhancement: allow callers to specify the list of SnapshotRetrieveIncludeModel to retrieve
-    getRepositoryApi(accessToken).retrieveSnapshot(snapshotId, java.util.Collections.emptyList())
+    addDataRepoRowId(getRepositoryApi(accessToken).retrieveSnapshot(snapshotId, java.util.Collections.emptyList()))
+  }
+
+  //Snapshots-by-reference always have a datarepo_row_id, but that is not included in the model and should be
+  private def addDataRepoRowId(snapshot: SnapshotModel): SnapshotModel = {
+    if (snapshot.getTables != null){
```

Comment:
This is null in the case of mocking, so tests will fail without the null check - seems reasonable to have one, although maybe it would be better to just update the mock to not have null tables?
Also, since snapshot's tables is a Java list, it's not easy to do a scala-ish null check.  Would welcome suggestions on a better way to do this.

Reply:
Perhaps a cleaner way:
```scala
    Option(snapshot.getTables) foreach { tables =>
      snapshot.tables(tables.stream().map(t => t.addColumnsItem(datareporow_id)).collect(Collectors.toList()))
    }
    snapshot
```